index.html

```diff
@@ -367,45 +368,64 @@ <h3>
           return [=a promise rejected with=] {{"NotAllowedError"}}
           {{DOMException}}.
           </li>
+          <li>Let |record| be the <a>platform wake lock</a>'s <a>state
+          record</a> associated with |document| and |type|.
+          </li>
           <li>Let |promise:Promise| be [=a new promise=].
           </li>
-          <li>Return |promise| and run the following steps <a>in parallel</a>:
+          <li>Run the following steps <a>in parallel</a>:
             <ol>
-              <li>
-                <a>Abort when</a> the steps to <a>determine the visibility
-                state</a> return `hidden`.
-              </li>
-              <li>Let |state:PermissionState| be the result of awaiting
-              <a>obtain permission</a> steps with "`screen-wake-lock`":
+              <li>Let |state:PermissionState| be the result of invoking
+              <a>obtain permission</a> with "`screen-wake-lock`":
                 <ol>
                   <li data-tests="wakelock-request-denied.https.html">If
-                  |state| is {{PermissionState/"denied"}}, then reject
-                  |promise| with a {{"NotAllowedError"}} {{DOMException}}, and
-                  abort these steps.
+                  |state| is {{PermissionState/"denied"}}, then <a>queue a
+                  global task</a> on the <a>screen wake lock task source</a>
+                  given |document|'s <a>relevant global object</a> to reject
+                  |promise| with a {{"NotAllowedError"}} {{DOMException}}
+                  </li>
+                  <li>Abort these steps.
```

Comment:
This aborts no matter what. You probably want to nest this underneath the "If...denied" as a sibling of "Queue a global task".

Reply:
Oops, you're right. I tried to incorporate some of the changes from #307 but didn't pay attention to this one. Fixed in #308 and here.